type error in ts

**Describe the bug**
```
error TS2322: Type 'ImageTileSource' is not assignable to type 'DataTileSource<DataTile>'.
  The types returned by 'getReprojTile_(...)' are incompatible between these types.
    Type 'ImageTile' is missing the following properties from type 'DataTile': loader_, data_, error_, size_, and 4 more.
```



**To Reproduce**
```ts
new WebGLTileLayer({
    source: new ImageTileSource({
        url: props.url,
        wrapX: true,
}),
```

**Expected behavior**
use WebGLTileLayer to render image tile

--- a/src/ol/layer/WebGLTile.js
+++ b/src/ol/layer/WebGLTile.js
@@ -17,7 +17,7 @@ import {
 import {expressionToGlsl} from '../webgl/styleparser.js';
 
 /**
- * @typedef {import("../source/DataTile.js").default} SourceType
+ * @typedef {import("../source/DataTile.js").default<import("../DataTile.js").default|import("../ImageTile.js").default>} SourceType
  */
 
 /**

--- a/test/typescript/cases/webgl-tile-layer.ts
+++ b/test/typescript/cases/webgl-tile-layer.ts
@@ -1,0 +1,9 @@
+import ImageTileSource from '../../../src/ol/source/ImageTile.js';
+import WebGLTileLayer from '../../../src/ol/layer/WebGLTile.js';
+
+new WebGLTileLayer({
+  source: new ImageTileSource({
+    url: 'http://example.com/tiles/{z}/{x}/{y}.png',
+    wrapX: true,
+  }),
+});

